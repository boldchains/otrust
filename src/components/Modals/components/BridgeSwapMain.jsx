import React, { useState, useEffect, useCallback, useMemo } from 'react';
import { useWeb3React } from '@web3-react/core';
import { BigNumber } from 'bignumber.js';
import cosmos from 'cosmos-lib';
import { ethers } from 'ethers';
import { useMediaQuery } from 'react-responsive';

import { useKeplr } from 'hooks/useKeplr';
import { useChain } from 'context/chain/ChainContext';
import { GravityCont, NOMCont } from 'context/chain/contracts';
import BridgeSwapMobile from './BridgeSwapMobile';
import BridgeSwapModal from './BridgeSwapModal';
import { contAddrs } from '../../../context/chain/contracts';
import { NOTIFICATION_MESSAGES } from '../../../constants/NotificationMessages';
import { responsive } from 'theme/constants';
import { useGasPriceSelection } from 'hooks/useGasPriceSelection';

export const initialErrorsState = { amountError: '', onomyWalletError: '', transactionError: '' };

export const initialGasOptions = [
  {
    id: 0,
    text: '00.00 (Standard)',
  },
  {
    id: 1,
    text: '00.00 (Fast)',
  },
  {
    id: 2,
    text: '00.00 (Instant)',
  },
];

export default function BridgeSwapMain({ closeBridgeModal }) {
  const keplrWallet = useKeplr();
  const [onomyWalletValue, setOnomyWalletValue] = useState('');
  const [amountValue, setAmountValue] = useState('');
  const [errors, setErrors] = useState(initialErrorsState);
  const [formattedWeakBalance, setFormattedWeakBalance] = useState(0);
  const [isDisabled, setIsDisabled] = useState(false);
  const [isTransactionPending, setIsTransactionPending] = useState(false);
  const [allowanceAmountGravity, setAllowanceAmountGravity] = useState(0);
  const [showBridgeExchangeModal, setShowBridgeExchangeModal] = useState(true);
  const [showApproveModal, setShowApproveModal] = useState(false);
  const [showTransactionCompleted, setShowTransactionCompleted] = useState(false);
  const [showLoader, setShowLoader] = useState(false);
  const { gasPriceChoice, setGasPriceChoice, gasOptions, gasPrice } = useGasPriceSelection();

  const standardBrigdeBreakpoint = useMediaQuery({ minWidth: responsive.smartphoneLarge });

  const { account, library } = useWeb3React();
  const { weakBalance } = useChain();

  const GravityContract = useMemo(() => GravityCont(library), [library]);
  const NOMContract = useMemo(() => NOMCont(library), [library]);

  useEffect(() => {
    if (!keplrWallet) return;
    console.log('keplrWallet', keplrWallet);
    const [acct] = keplrWallet.accounts;
    setOnomyWalletValue(acct.address);
  }, [keplrWallet]);

  useEffect(() => {
    setFormattedWeakBalance(weakBalance.shiftedBy(-18));
  }, [weakBalance]);

  const updateAllowanceAmount = useCallback(async () => {
    const allowanceGravity = await NOMContract.allowance(account, contAddrs.Gravity);
    setAllowanceAmountGravity(allowanceGravity);
    return allowanceGravity;
  }, [NOMContract, account]);

  useEffect(() => {
    if (NOMContract && account && !allowanceAmountGravity) {
      updateAllowanceAmount();
    }
  }, [NOMContract, account, allowanceAmountGravity, updateAllowanceAmount]);

  const walletChangeHandler = event => {
    setOnomyWalletValue(event.target.value);
  };

  const amountChangeHandler = event => {
    const { value } = event.target;
    const floatRegExp = new RegExp(
      /(^(?=.+)(?:[1-9]\d*|0)?(?:\.\d{1,18})?$)|(^\d+?\.$)|(^\+?(?!0\d+)$|(^$)|(^\.$))/
    );
    if (floatRegExp.test(value)) {
      setAmountValue(value);
      const bigAmountShifted18 = BigNumber(value).shiftedBy(18);
      if (bigAmountShifted18.gt(weakBalance)) {
        setErrors(prevState => {
          return { ...prevState, amountError: NOTIFICATION_MESSAGES.error.insufficientFunds };
        });
        setIsDisabled(true);
      } else {
        setErrors(prevState => {
          return { ...prevState, amountError: '' };
        });
        setIsDisabled(false);
      }
    }
  };

  const maxBtnClickHandler = event => {
    event.preventDefault();
    if (weakBalance.toNumber()) {
      setAmountValue(formattedWeakBalance.toString(10));
      setIsDisabled(false);
    }
  };

  const onCancelClickHandler = () => {
    setShowApproveModal(false);
    setShowBridgeExchangeModal(true);
    setIsDisabled(false);
    updateAllowanceAmount();
  };

  const closeModal = () => {
    if (!isTransactionPending) {
      closeBridgeModal();
    }
  };

  const submitTransClickHandler = useCallback(
    async event => {
      let cosmosAddressBytes32;
      event.preventDefault();
      setErrors(initialErrorsState);
      if (amountValue === '.' || !parseFloat(amountValue)) {
        setErrors(prevState => {
          return { ...prevState, amountError: NOTIFICATION_MESSAGES.error.incorrectAmountFormat };
        });
        return;
      }

      const string18FromAmount = BigNumber(amountValue).shiftedBy(18).toString(10);

      try {
        setIsDisabled(true);
        const bytes = cosmos.address.getBytes(onomyWalletValue);
        const ZEROS = Buffer.alloc(12);
        cosmosAddressBytes32 = Buffer.concat([ZEROS, bytes]);
      } catch (error) {
        setErrors(prevState => {
          return {
            ...prevState,
            onomyWalletError: NOTIFICATION_MESSAGES.error.incorrectOnomyAddressFormat,
          };
        });
        setIsDisabled(false);
        return;
      }

      let tx;
      if (allowanceAmountGravity.gte(ethers.BigNumber.from(string18FromAmount))) {
        try {
          setShowLoader(true);
          setIsTransactionPending(true);
          tx = await GravityContract.sendToCosmos(
            contAddrs.NOMERC20,
            cosmosAddressBytes32,
            string18FromAmount,
            {
              gasPrice: gasPrice.toFixed(0),
            }
          );

          tx.wait().then(() => {
            setIsDisabled(false);
            setShowBridgeExchangeModal(false);
            setShowTransactionCompleted(true);
            setShowLoader(false);
            setIsTransactionPending(false);
          });
        } catch (error) {
          if (error.code === 4001) {
            setErrors(prevState => {
              return {
                ...prevState,
                transactionError: NOTIFICATION_MESSAGES.error.rejectedTransaction,
              };
            });
          } else {
            setErrors(prevState => {
              return { ...prevState, transactionError: error.message };
            });
          }

          setShowLoader(false);
          setIsDisabled(false);
          setIsTransactionPending(false);
        }
      } else {
        setShowBridgeExchangeModal(false);
        setShowApproveModal(true);
      }
    },
    [onomyWalletValue, amountValue, GravityContract, allowanceAmountGravity, gasPrice]
  );

  const Props = {
    values: {
      onomyWalletValue,
      amountValue,
      formattedWeakBalance,
      allowanceAmountGravity,
      weakBalance,
      errors,
      gasPrice,
      gasPriceChoice,
      gasOptions,
    },
    flags: {
      isDisabled,
      isTransactionPending,
      showBridgeExchangeModal,
      showApproveModal,
      showTransactionCompleted,
      showLoader,
    },
    handlers: {
      walletChangeHandler,
      amountChangeHandler,
      maxBtnClickHandler,
      submitTransClickHandler,
      onCancelClickHandler,
      closeModal,
      setGasPriceChoice,
    },
  };

  return (
    <>
      {standardBrigdeBreakpoint ? <BridgeSwapModal {...Props} /> : <BridgeSwapMobile {...Props} />}
    </>
  );
}
