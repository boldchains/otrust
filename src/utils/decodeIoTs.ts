import { isLeft } from 'fp-ts/lib/Either';
import * as t from 'io-ts';

export function decodeIoTs<T, O, I>(validator: t.Type<T, O, I>, input: I): T {
  const result = validator.decode(input);

  if (isLeft(result)) {
    throw new Error(result.left.toString());
  }

  return input as unknown as T;
}
