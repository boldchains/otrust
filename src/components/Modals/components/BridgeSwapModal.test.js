import { cleanup } from '@testing-library/react';
import { BigNumber } from 'bignumber.js';

import { initialGasOptions, initialErrorsState } from './BridgeSwapMain';
import BridgeSwapModal from './BridgeSwapModal';
import { renderWithContext } from '../../../utils/testing';

const testProps = {
  values: {
    onomyWalletValue: '',
    amountValue: '',
    formattedWeakBalance: BigNumber(0),
    allowanceAmountGravity: BigNumber(0),
    weakBalance: BigNumber(0),
    errors: initialErrorsState,
    gasPrice: 0,
    gasPriceChoice: '',
    gasOptions: initialGasOptions,
  },
  flags: {
    isDisabled: false,
    isTransactionPending: false,
    showBridgeExchangeModal: true,
    showApproveModal: false,
    showTransactionCompleted: false,
    showLoader: false,
  },
  handlers: {
    walletChangeHandler: jest.fn(),
    amountChangeHandler: jest.fn(),
    maxBtnClickHandler: jest.fn(),
    submitTransClickHandler: jest.fn(),
    onCancelClickHandler: jest.fn(),
    closeModal: jest.fn(),
    setGasPriceChoice: jest.fn(),
    setGasPrice: jest.fn(),
  },
};

describe('Given the BridgeSwapModal component', () => {
  describe('when the component is rendered', () => {
    afterEach(cleanup);

    it('should match the snapshot', () => {
      const { asFragment } = renderWithContext(BridgeSwapModal, testProps);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
