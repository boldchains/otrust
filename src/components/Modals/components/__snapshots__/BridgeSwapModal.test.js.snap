// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the BridgeSwapModal component when the component is rendered should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  background-color: rgba(0,0,0,0.7);
  z-index: 10;
}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:480px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:480px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:480px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:700px) {

}

@media screen and (max-width:1180px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:700px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:834px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:834px) {

}

@media screen and (max-width:834px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1180px) {

}

@media screen and (max-width:700px) {

}

@media screen and (max-width:834px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1180px) {

}

@media screen and (max-width:834px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1180px) {

}

@media screen and (max-width:834px) {

}

@media screen and (max-width:1440px) {

}

@media screen and (max-width:1180px) {

}

<div
    class="c0"
  />
  .c0 {
  width: 500px;
  padding: 4px;
  position: absolute;
  top: 50%;
  left: 50%;
  background-color: #201d2a;
  border-radius: 8px;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 11;
}

.c0 main {
  padding: 32px;
}

.c0 footer {
  padding: 32px 36px 0;
  background-color: #1a1823;
}

.c2 {
  width: 24px;
  height: 24px;
  position: absolute;
  top: 10px;
  right: 10px;
  background: transparent;
  border: none;
}

.c2 svg {
  fill: #656273;
}

.c2:hover svg {
  fill: #9895a6;
}

.c2:active svg {
  fill: #4d4b57;
}

.c25 {
  width: 170px;
  height: 50px;
  background-color: #302E3D;
  border: none;
  border-radius: 8px;
  color: #656273;
  font-weight: 600;
  font-size: 14px;
}

.c25:hover {
  background-color: #302E3D;
}

.c25:active {
  background-color: #302E3D;
}

.c4 {
  margin: 0 0 24px;
  font-size: 16px;
  font-weight: 500;
  text-align: center;
}

.c1 {
  width: 1300px;
}

.c27 {
  padding: 40px;
  border-radius: 8px;
  background-color: #1a1823;
}

.c27 h2 {
  margin: 0 0 20px;
  font-family: Bebas Neue,sans-serif;
  font-size: 28px;
  font-weight: 300;
}

.c30 {
  display: grid;
  grid-template-columns: 1fr 180px;
  gap: 30px;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  margin-bottom: 10px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 24px;
  padding: 20px;
  border-radius: 8px;
  background-color: #1a1823;
}

.c33 {
  padding: 40px 50px;
  border-radius: 8px;
  background-color: #201d2a;
}

.c31 {
  margin: 0 0 12px;
  font-size: 16px;
  font-weight: 500;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 56px;
  min-width: 56px;
  height: 56px;
  border-radius: 50%;
  border: 3px solid #302E3D;
  background-color: #201d2a;
}

.c8 img {
  overflow: hidden;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 8px;
}

.c9 > strong {
  font-weight: 500;
  white-space: nowrap;
}

.c9 > span {
  font-size: 16px;
  font-weight: 500;
  color: #9895a6;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
  margin-left: auto;
}

.c10 > strong {
  font-size: 12px;
  font-weight: 400;
  color: #9895a6;
  white-space: nowrap;
}

.c10 span {
  font-family: Bebas Neue,sans-serif;
  font-size: 28px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 40px auto;
  padding: 16px 40px;
  position: relative;
  border-radius: 8px;
  background-color: #30232e;
  color: #c75a5a;
  white-space: nowrap;
}

.c11::before,
.c11::after {
  content: '';
  display: block;
  width: 1px;
  height: 20px;
  position: absolute;
  top: -30px;
  left: 50%;
  background-color: #c75a5a;
}

.c11::after {
  top: auto;
  bottom: -30px;
  background-color: #c75a5a;
}

.c3 {
  display: grid;
  grid-template-columns: 520px 1fr;
}

.c3 > main {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  width: 100%;
  border: 1px solid #302E3D;
  border-radius: 8px;
  padding: 16px 29px 16px 20px;
  opacity: 1;
}

.c29 {
  max-width: 600px;
  margin-bottom: 40px;
  color: #9895a6;
}

.c29 strong {
  font-weight: 500;
  color: #e1dfeb;
}

.c32 {
  padding-left: 1em;
  color: #9895a6;
}

.c32 li::marker {
  color: #e1dfeb;
}

.c32 li + li {
  margin-top: 12px;
}

.c26 {
  width: 100%;
  height: 52px;
  background-color: #302E3D;
  color: #656273;
}

.c26:hover {
  background-color: #302E3D;
}

.c26:active {
  background-color: #302E3D;
}

.c5 {
  text-align: left;
}

.c6 {
  display: grid;
  grid-template-rows: repeat(3,auto);
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 70px;
  padding: 16px 8px 16px 16px;
  background-color: #302E3D;
  border-radius: 8px;
  border-bottom: 2px solid #1a1823;
  color: #9895a6;
  font-size: 14px;
  font-weight: 500;
}

.c17 strong {
  color: #9895a6;
  margin-right: auto;
  font-weight: 500;
  white-space: nowrap;
}

.c18 {
  background: transparent;
  border: 1px solid #302E3D;
}

.c20 {
  padding: 15px 20px;
  height: 54px;
  border: none;
  background-color: #201d2a;
  border-radius: 7px;
  color: #ffdda1;
  font-weight: 500;
  font-size: 14px;
  text-transform: uppercase;
  cursor: pointer;
}

.c20:hover {
  background-color: #252130;
}

.c20:active {
  background-color: #1b1924;
}

.c21 {
  background-color: transparent;
}

.c13 {
  width: 100%;
  background: none;
  border: none;
  color: #e1dfeb;
  font-weight: 500;
  font-size: 18px;
  text-align: right;
}

.c13:focus {
  outline: none;
}

.c14 {
  width: 100%;
  text-align: left;
  font-size: 14px;
}

.c19 {
  width: 100%;
  padding: 0 10px;
}

.c16 {
  margin: 0 0 12px;
}

.c15 {
  padding: 32px 32px 0;
  margin: 32px -32px 0 -36px;
  border-top: 1px solid #302E3D;
}

.c22 {
  margin: 0 0 12px;
  color: #656273;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  gap: 12px;
  margin: 12px 0 16px;
}

.c24 {
  padding: 12px 16px;
  background-color: transparent;
  border: 1px solid #302E3D;
  border-radius: 22px;
  font-size: 14px;
  font-weight: 500;
  color: #9895a6;
}

.c24:hover {
  background-color: #47445a;
}

.c24:active {
  background-color: #191820;
}

.c28 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (max-width:1440px) {
  .c0 {
    width: 400px;
  }
}

@media screen and (max-width:480px) {
  .c0 {
    width: 100%;
    border-radius: 0;
    top: unset;
    left: 0;
    bottom: 0;
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

@media screen and (max-width:1440px) {
  .c0 main {
    padding: 24px;
  }
}

@media screen and (max-width:1440px) {
  .c0 footer {
    padding: 24px 24px 0;
  }
}

@media screen and (max-width:1440px) {
  .c25 {
    width: 135px;
    height: 40px;
    font-size: 12px;
  }
}

@media screen and (max-width:480px) {
  .c25 {
    font-size: 14px;
  }
}

@media screen and (max-width:1440px) {
  .c4 {
    font-size: 14px;
  }
}

@media screen and (max-width:480px) {
  .c4 {
    font-size: 16px;
  }
}

@media screen and (max-width:1440px) {
  .c1 {
    width: 1010px;
  }
}

@media screen and (max-width:700px) {
  .c1 {
    display: none;
  }
}

@media screen and (max-width:1180px) {
  .c1 {
    width: 600px;
    top: 20px;
    -webkit-transform: translate(-50%,0);
    -ms-transform: translate(-50%,0);
    transform: translate(-50%,0);
  }
}

@media screen and (max-width:1440px) {
  .c27 {
    padding: 24px;
  }
}

@media screen and (max-width:1440px) {
  .c27 h2 {
    margin-bottom: 16px;
    font-size: 24px;
  }
}

@media screen and (max-width:1440px) {
  .c30 {
    grid-template-columns: 1fr 150px;
  }
}

@media screen and (max-width:700px) {
  .c30 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c30 > img {
    width: 230px;
    margin: 32px auto;
  }
}

@media screen and (max-width:1440px) {
  .c7 {
    gap: 20px;
  }
}

@media screen and (max-width:834px) {
  .c7:first-child {
    grid-column: 1/3;
  }
}

@media screen and (max-width:1440px) {
  .c33 {
    padding: 40px;
  }
}

@media screen and (max-width:1440px) {
  .c31 {
    font-size: 14px;
  }
}

@media screen and (max-width:1440px) {
  .c8 {
    width: 48px;
    min-width: 48px;
    height: 48px;
  }
}

@media screen and (max-width:1440px) {
  .c9 > span {
    font-size: 14px;
  }
}

@media screen and (max-width:1440px) {
  .c10 > strong {
    font-size: 10px;
  }
}

@media screen and (max-width:1440px) {
  .c10 span {
    font-size: 24px;
  }
}

@media screen and (max-width:834px) {
  .c11 {
    grid-column: 2/3;
    grid-row: 2/3;
    height: 100%;
    margin: 0;
    padding: 24px;
    white-space: pre-wrap;
    text-align: center;
  }
}

@media screen and (max-width:834px) {
  .c11::before,
  .c11::after {
    display: none;
  }
}

@media screen and (max-width:1440px) {
  .c3 {
    grid-template-columns: 400px 1fr;
  }
}

@media screen and (max-width:1180px) {
  .c3 {
    grid-template-columns: none;
  }
}

@media screen and (max-width:700px) {
  .c12 {
    margin-top: 16px;
  }
}

@media screen and (max-width:834px) {
  .c6 {
    grid-template-columns: 1fr 150px;
    grid-template-rows: 1fr 1fr;
    gap: 24px;
  }
}

@media screen and (max-width:1440px) {
  .c17 {
    height: 60px;
    padding: 16px 8px;
  }
}

@media screen and (max-width:1180px) {
  .c17 {
    height: 55px;
    padding: 10px 5px 10px 8px;
    font-size: 12px;
  }
}

@media screen and (max-width:834px) {
  .c17 strong {
    font-size: 12px;
  }
}

@media screen and (max-width:1440px) {
  .c20 {
    height: 44px;
  }
}

@media screen and (max-width:1180px) {
  .c13 {
    font-size: 14px;
  }
}

@media screen and (max-width:834px) {
  .c13 {
    font-size: 12px;
  }
}

@media screen and (max-width:1440px) {
  .c15 {
    margin-left: -28px;
    margin-right: -24px;
  }
}

@media screen and (max-width:1180px) {
  .c15 {
    margin-left: -28px;
    margin-right: -28px;
  }
}

<div
    class="c0 c1"
  >
    <button
      class="c2"
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          opacity="0.5"
        >
          <g
            transform="translate(-.453 .555)"
          >
            <path
              class="cls-3"
              d="M8.762.384a1.311 1.311 0 011.981 1.707l-.127.147-8.378 8.378A1.311 1.311 0 01.257 8.909l.127-.147z"
              transform="translate(6.453 6.445)"
            />
            <path
              class="cls-3"
              d="M.384.384A1.311 1.311 0 012.09.257l.147.127 8.379 8.38a1.31 1.31 0 01-1.706 1.98l-.147-.127L.384 2.236a1.31 1.31 0 010-1.852z"
              transform="translate(6.453 6.445)"
            />
          </g>
        </g>
      </svg>
    </button>
    <div
      class="c3"
    >
      <main>
        <h3
          class="c4 c5"
        >
          Onomy Bridge Here
        </h3>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <img
                alt=""
                src="icon-bridge-curve.svg"
              />
            </div>
            <div
              class="c9"
            >
              <strong>
                Onomy Bonding Curve
              </strong>
              <span />
            </div>
            <div
              class="c10"
            >
              <strong>
                wNOM Balance
              </strong>
              <span>
                0.000000
              </span>
            </div>
          </div>
          <div
            class="c11"
          >
            Wallet Disconnected
          </div>
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <img
                alt=""
                src="icon-onomy-wallet.svg"
              />
            </div>
            <div
              class="c12"
            >
              <input
                class="c13 c14"
                placeholder="Your Onomy Wallet Address"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <form
          class="c15"
        >
          <div
            class="c16"
          >
            <div
              class="c17 c18"
            >
              <strong>
                Swap to NOM
              </strong>
              <input
                class="c13 c19"
                type="text"
                value=""
              />
              wNOM
              <button
                class="c20 c21"
              >
                Max
              </button>
            </div>
          </div>
          <div>
            <p
              class="c22"
            >
              Gas Fee
            </p>
            <div
              class="c23"
            >
              <button
                class="c24"
              >
                00.00 (Standard)
              </button>
              <button
                class="c24"
              >
                00.00 (Fast)
              </button>
              <button
                class="c24"
              >
                00.00 (Instant)
              </button>
            </div>
          </div>
          <button
            class="c25 c26"
            disabled=""
          >
            Swap wNOM for NOM
          </button>
        </form>
      </main>
      <aside
        class="c27"
      >
        <header
          class="c28"
        >
          <h2>
            What is Onomy Bridge?
          </h2>
        </header>
        <p
          class="c29"
        >
          The Onomy Bonding Curve platform is a gateway into the Onomy Network. This is achieved by participants purchasing wrapped-NOM, an ERC-20 token on the Ethereum Network, and swapping for NOM on the Onomy Network.
        </p>
        <section
          class="c30"
        >
          <div>
            <h3
              class="c31"
            >
              One Way Bridge
            </h3>
            <p
              class="c29"
            >
              Choose to bridge when you are ready to do so to finalize your purchase of NOM! 
              <strong>
                After bridging, you can no longer sell back to the bonding curve or bridge back for wNOM.
              </strong>
               There are no guarantees of liquid markets.
            </p>
          </div>
          <img
            alt=""
            src="one-way-bridge.svg"
          />
        </section>
        <section
          class="c30"
        >
          <div>
            <h3
              class="c31"
            >
              Why Bridge?
            </h3>
            <ol
              class="c32"
            >
              <li>
                You must hold NOM to participate in the Onomy Network. 
              </li>
              <li>
                Early stakers of NOM take advantage of larger staking yield. 
              </li>
              <li>
                NOM is used for governance, staking, and collateral to mint stablecoins.
              </li>
              <li>
                All bridged wNOM is burned from the bonding curve supply. 
              </li>
              <li>
                NOM would be listed on exchanges rather than wNOM. 
              </li>
            </ol>
          </div>
          <div
            class="c33"
          >
            <img
              alt=""
              src="why-bridge.svg"
            />
          </div>
        </section>
      </aside>
    </div>
  </div>
</DocumentFragment>
`;
